aws/resource_aws_efs_access_point.go

```diff
@@ -0,0 +1,404 @@
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("file-system/%s", aws.StringValue(ap.FileSystemId)),
+		Service:   "elasticfilesystem",
+	}.String()
+
+	d.Set("file_system_arn", fsARN)
+	d.Set("file_system_id", ap.FileSystemId)
+	d.Set("arn", ap.AccessPointArn)
+	d.Set("owner_id", ap.OwnerId)
+
+	if err := d.Set("posix_user", flattenEfsAccessPointPosixUser(ap.PosixUser)); err != nil {
+		return fmt.Errorf("error setting posix user: %s", err)
+	}
+
+	if err := d.Set("root_directory", flattenEfsAccessPointRootDirectory(ap.RootDirectory)); err != nil {
+		return fmt.Errorf("error setting root directory: %s", err)
+	}
+
+	if err := d.Set("tags", keyvaluetags.EfsKeyValueTags(ap.Tags).IgnoreAws().Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsEfsAccessPointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	log.Printf("[DEBUG] Deleting EFS access point %q", d.Id())
+	_, err := conn.DeleteAccessPoint(&efs.DeleteAccessPointInput{
+		AccessPointId: aws.String(d.Id()),
+	})
+	if err != nil {
+		return err
+	}
+
+	err = waitForDeleteEfsAccessPoint(conn, d.Id(), 10*time.Minute)
+	if err != nil {
+		return fmt.Errorf("Error waiting for EFS access point (%q) to delete: %s", d.Id(), err.Error())
+	}
+
+	log.Printf("[DEBUG] EFS access point %q deleted.", d.Id())
+
+	return nil
+}
+
+func waitForDeleteEfsAccessPoint(conn *efs.EFS, id string, timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{efs.LifeCycleStateAvailable, efs.LifeCycleStateDeleting, efs.LifeCycleStateDeleted},
+		Target:  []string{},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := conn.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+				AccessPointId: aws.String(id),
+			})
+			if err != nil {
+				if isAWSErr(err, efs.ErrCodeAccessPointNotFound, "") {
+					return nil, "", nil
+				}
+
+				return nil, "error", err
+			}
+
+			if hasEmptyAccessPoints(resp) {
+				return nil, "", nil
+			}
+
+			mt := resp.AccessPoints[0]
+
+			log.Printf("[DEBUG] Current status of %q: %q", *mt.AccessPointId, *mt.LifeCycleState)
+			return mt, *mt.LifeCycleState, nil
+		},
+		Timeout:    timeout,
+		Delay:      2 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func hasEmptyAccessPoints(aps *efs.DescribeAccessPointsOutput) bool {
+	if aps != nil && len(aps.AccessPoints) > 0 {
+		return false
+	}
+	return true
+}
+
+func expandEfsAccessPointPosixUser(pUser []interface{}) *efs.PosixUser {
+	if len(pUser) < 1 || pUser[0] == nil {
+		return nil
+	}
+
+	m := pUser[0].(map[string]interface{})
+
+	posixUser := &efs.PosixUser{
+		Gid: aws.Int64(int64(m["gid"].(int))),
+		Uid: aws.Int64(int64(m["uid"].(int))),
+	}
+
+	if v, ok := m["secondary_gids"]; ok && len(v.(*schema.Set).List()) > 0 {
+		posixUser.SecondaryGids = expandInt64Set(v.(*schema.Set))
+	}
```

Comment:
Neat Go fact, its safe to perform map lookup and type assertion at the same time when using the second return value check, which can save some logic:

```suggestion
	if v, ok := m["secondary_gids"].(*schema.Set); ok && len(v.List()) > 0 {
		posixUser.SecondaryGids = expandInt64Set(v)
	}
```